api/pkg/transformer/feast/transformer.go

```diff
@@ -86,13 +89,24 @@ func NewTransformer(feastClient feast.Client, config *transformer.StandardTransf
 	}
 
 	compiledJsonPath := make(map[string]*jsonpath.Compiled)
+	compiledUdf := make(map[string]*vm.Program)
 	for _, ft := range config.TransformerConfig.Feast {
 		for _, configEntity := range ft.Entities {
-			c, err := jsonpath.Compile(configEntity.JsonPath)
-			if err != nil {
-				return nil, fmt.Errorf("unable to compile jsonpath for entity %s: %s", configEntity.Name, configEntity.JsonPath)
+			switch configEntity.Extractor.(type) {
+			case *transformer.Entity_JsonPath:
+				c, err := jsonpath.Compile(configEntity.GetJsonPath())
+				if err != nil {
+					return nil, fmt.Errorf("unable to compile jsonpath for entity %s: %s", configEntity.Name, configEntity.GetJsonPath())
+				}
+				compiledJsonPath[configEntity.GetJsonPath()] = c
+			case *transformer.Entity_Udf:
+				c, err := expr.Compile(configEntity.GetUdf(), expr.Env(UdfEnv{}))
+				if err != nil {
+					return nil, err
+				}
+				compiledUdf[configEntity.Name] = c
```

Comment:
Using entity name as the map key might cause issue when there are same entity but in different retrieval table.